system/jlib/jlz4.cpp

```diff
@@ -221,6 +222,60 @@ class jlib_decl CLZ4Expander : public CFcmpExpander
         }
     }
 
+    size32_t expandFirst(MemoryBuffer & target, const void * src)
+    {
+        init(src);
+        totalExpanded = 0;
+        return expandNext(target);
+    }
+
+    size32_t expandNext(MemoryBuffer & target)
+    {
+        if (totalExpanded == outlen)
+            return 0;
+
+        const size32_t szchunk = *in;
+        in++;
+
+        target.clear();
+        size32_t written;
+        if (szchunk+totalExpanded<outlen)
+        {
+            size32_t maxOut = target.capacity();
+            size32_t maxEstimate = (outlen - totalExpanded);
+            size32_t estimate = szchunk; // possible * 2/5 but don't want to expand too much
+            if (estimate > maxEstimate)
```

Comment:
is this related to there potentially being an uncompressed trailer?

Reply:
no, this is because you don't know how big the decompressed block will be.  In the spill file case we do actually know - it should be <= the block size so the target is preallocated to that size.